### ISSUE TITLE
Yolov8-obb found 134 images but is only training on 9 images.

### ISSUE THREAD

**Issue by AhmedHamadto** [ultralytics/ultralytics#18067]
Hello ultralytics,
I am trying to train an oriented bounding box object detection model using yolov8 and yolov11, I have received the exact issue whilst trying on both.

I would appreciate some help in fixing this issue.

### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

train: Scanning /home/innoadmin/OBB-Hang-up/train/labels.cache... 134 images, 0 backgrounds, 0 corrupt: 100%|██████████|
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))

val: Scanning /home/innoadmin/OBB-Hang-up/val/labels.cache... 18 images, 0 backgrounds, 0 corrupt: 100%|██████████| 18/1
Plotting labels to runs/obb/train8/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 87 weight(decay=0.0), 97 weight(decay=0.0005), 96 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 1024 train, 1024 val
Using 8 dataloader workers
Logging results to runs/obb/train8
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      6.33G      0.994      1.157      2.213          9       1024: 100%|██████████| 9/9 [00:02<00:00,  3.19i
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00
                   all         18         18      0.986          1      0.995      0.738

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/100      5.88G     0.9734     0.7791      2.164         14       1024: 100%|██████████| 9/9 [00:01<00:00,  7.26i
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00
                   all         18         18      0.995      0.944      0.992      0.735

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/100      5.89G     0.9237     0.7111      2.156         12       1024: 100%|██████████| 9/9 [00:01<00:00,  7.25i
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00
                   all         18         18      0.997          1      0.995      0.752

### Environment

Ultralytics 8.3.4 🚀 Python-3.10.12 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 4090, 24118MiB)
Setup complete ✅ (16 CPUs, 62.4 GB RAM, 875.2/877.2 GB disk)

OS                  Linux-5.15.0-125-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
RAM                 62.42 GB
CPU                 Intel Xeon Silver 4309Y 2.80GHz
CUDA                11.7

numpy               ✅ 1.23.5>=1.23.0
matplotlib          ✅ 3.7.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 9.4.0>=7.1.2
pyyaml              ✅ 5.4.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 2.0.0>=1.8.0
torchvision         ✅ 0.15.1>=0.9.0
tqdm                ✅ 4.65.0>=4.64.0
psutil              ✅ 5.9.4
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 1.5.3>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0
ultralytics-thop    ✅ 2.0.12>=2.0.0
torch               ✅ 2.0.0!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

from ultralytics import YOLO

model = YOLO("yolov8n-obb.pt")
results = model.train(data="data.yaml", epochs=100)

This is the yaml file:
path: ../../OBB-Hang-up
train: train/images/
val: val/images/
test: test/images/

nc: 1
names: ['connector']

This is the pwd:
/home/innoadmin/OBB-Hang-up

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AhmedHamadto, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. For example, include a detailed dataset config file (`data.yaml`), any custom code/scripts, or specific commands that reproduce the issue. This will allow us to better understand what's going on and isolate the problem faster. If applicable, also ensure that your dataset's image-label matching is correct.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples, distribution of labels, and training logs, and verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help guide you. An Ultralytics engineer will review your issue and assist you further as soon as possible. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
9 in the logs refer to 9 batches. Not 9 images. Each batch has upto 16 images by default unless you change it with `batch` argument.

**Comment by AhmedHamadto** [comment]
Thanks a bunch, mind pointing me to where you found this out as the normal model shows number of images not batches.

**Comment by Farenweh** [comment]
Smallest training step is an Iteration, which contains Forward-Loss-Backward-Update on a batch of images, thus you could find any other statistic changes per Iter. So, the bar indicates batches.

**Comment by glenn-jocher** [comment]
You're correct; the progress bar displays batches rather than images. This is because each training iteration processes a batch of images, as described in [Ultralytics' documentation on batch size](https://www.ultralytics.com/glossary/batch-size).
